troubadix/plugins/multiple_re_parameters.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2024 Greenbone AG
+
+import re
+from pathlib import Path
+from typing import Iterable, Iterator
+
+from troubadix.helper.patterns import (
+    SpecialScriptTag,
+    get_special_script_tag_pattern,
+)
+from troubadix.plugin import LineContentPlugin, LinterError, LinterResult
+
+re_pattern = re.compile(r"re\s*:")
+
+mandatory_keys_pattern = get_special_script_tag_pattern(
+    SpecialScriptTag.MANDATORY_KEYS
+)
+
+
+class CheckMultipleReParameters(LineContentPlugin):
+    """This step checks if a VT
+    has multiple mandatory_script_key re parameters
+    """
+
+    name = "check_multiple_re_parameters"
+
+    def check_lines(
+        self, nasl_file: Path, lines: Iterable[str]
+    ) -> Iterator[LinterResult]:
+
+        if self.context.nasl_file.suffix == ".inc":
+            return
+
+        matched_re = [
+            value
+            for line in lines
+            if not re.match(r"^\s*#", line)
+            if (match := mandatory_keys_pattern.search(line))
+            for value in re_pattern.findall(match.group("value"))
+        ]
```

Comment:
I'd prefer using a plain for-loop instead of two loops and if-statements inside a single list comprehension, took me a little too long to wrap my head around this :D

Maybe something like
```suggestion
        re_pattern_count = 0
        for line in lines:
            if not (match := mandatory_keys_pattern.search(line)):
                continue

            if re.match(r"^\s*#", line):
                continue

            re_pattern_count += len(re_pattern.findall(match.group("value")))
```

Reply:
amended in 78cfb2fcd7027dc7e3382c0185422fbd3d6fc6c6